game/src/Combat/Stats.gd

```diff
@@ -26,7 +26,12 @@ func _ready() -> void:
 func take_damage(hit: Hit) -> void:
 	if invulnerable:
 		return
-		
+
+	# Damages with value of -1 mean instakill
+	if hit.damage == -1:
```

Comment:
This is not intuitive, to the point you have to add a comment to explain it. You can have obstacles deal a lot of damage instead, like 100000.

Reply:
Or have a dedicated function, like `kill()` on the character and there. although I'd apply a lot of damage.

Reply:
I made this more explicit with an `is_instakill` bool on Hits and Damage sources.

I think we should refactor this system, it is too redundant, Hit just copies DamageSource properties.

We could make DamageSource use Hit, Hit being a `Resource`, and the HitBox catches this Hit upon overlapping with a DamageSource.

Reply:
> We could make DamageSource use Hit, Hit being a Resource, and the HitBox catches this Hit upon overlapping with a DamageSource.

The way I coded the hit was because I planned to add character stats to the game, and so the hit would store the result of calculations from the stats. At the moment I don't think it's worth changing as combat isn't really there. I think it'd be best to refactor it after working on the combat and progression in general, if that ever happens.